sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala

```diff
@@ -188,7 +196,7 @@ case class CreateDataSourceTableAsSelectCommand(
               throw new AnalysisException(s"Saving data in ${o.toString} is not supported.")
           }
         case SaveMode.Overwrite =>
-          sparkSession.sql(s"DROP TABLE IF EXISTS $tableName")
+          sessionState.catalog.dropTable(tableIdentWithDB, ignoreIfNotExists = false, purge = false)
```

Comment:
If ignoreIfNotExists = false, it throws an exception if the table doesn't exist.
```
// File SessionCatalog line 392
      } else if (!ignoreIfNotExists) {
        throw new NoSuchTableException(db = db, table = table)
      }
```